### PR TITLE
Close #6091: Create concept-crash interface in support-base.

### LINKED ISSUE #6091: Create concept-crash interface in support-base
Currently any components that wants an optional interaction with `lib-crash` forces consumer to include `lib-crash` even thought a `CrashReporter` reference was not created or passed it.  Create a `concept-crash` interface in `support-base` to decouple the `lib-crash` implementation from other components.

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: .buildconfig.yml [hunk 1]
```diff
@@ -32,6 +32,10 @@ projects:
     path: components/concept/sync
     description: 'An abstract definition of a browser data synchronization layer.'
     publish: true
+  concept-crash:
+    path: components/concept/crash
+    description: 'An abstract definition of a crash reporting component.'
+    publish: true
```
Comment (pocmo) [inline .buildconfig.yml]: Your commit message mentions `support-base`, but you created a separate concept component here. I think moving this into `support-base` is a good idea.

File: components/concept/crash/src/main/java/mozilla/components/concept/crash/ConceptCrashReporter.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.crash
+
+import kotlinx.coroutines.Job
+
+/**
+ * A concept crash reporter that can report caught exception to multiple services.
+ */
+interface ConceptCrashReporter {
```
Comment (pocmo) [inline components/concept/crash/src/main/java/mozilla/components/concept/crash/ConceptCrashReporter.kt]: Let's not use `Concept` in class names.

File: .buildconfig.yml (hunk 1 above)
Comment (rocketsroger) [reply .buildconfig.yml]: Sure, I'll move it.

**Comment by rocketsroger** [comment]
bors r=Amejia481

**Pull request merged**